src/Field/Definition/ChoiceField.php

```diff
@@ -95,31 +94,30 @@ public function validateData (ComponentContext $context, array $contentPath, mix
 
 	private function validateSingleSelect (ComponentContext $context, array $contentPath, mixed $data) : void
 	{
+		$data = $context->normalizeOptionalString($data);
+
+		if (null === $data)
+		{
+			return;
+		}
+
+		\assert(\is_string($data));
```

Comment:
This is obsolete now, as it is already determined by the return value of `normalizeOptionalString`.